fix(route/sjtu): gs

dzx-dzx opened the pull request:
<!-- 
如有疑问，请参考 https://docs.rsshub.app/joinus/new-rss/submit-route.html
Reference: https://docs.rsshub.app/en/joinus/new-rss/submit-route.html
-->

## 该 PR 相关 Issue / Involved Issue

Close #12742 

## 路由地址示例 / Example for the Proposed Route(s)

<!--
请在 `routes` 区域填写以 / 开头的完整路由地址，否则你的 PR 将会被无条件关闭。
如果路由包含在文档中列出可以完全穷举的参数（例如分类），请依次全部列出。

Please include route starts with /, with all required and optional parameters. Fail to comply will result in your pull request being closed automatically.
```route
```
如果你的 PR 与路由无关, 请在 `routes` 区域 填写 `NOROUTE`，而不是直接删除 `routes` 区域。否则你的 PR 将会被无条件关闭。
If your changes are not related to route, please fill in `routes` section with `NOROUTE`. Fail to comply will result in your PR being closed.
-->

```routes
/sjtu/gs/work
/sjtu/gs/enroll/58
/sjtu/gs/enroll/59
/sjtu/gs/enroll/60
/sjtu/gs/enroll/61
/sjtu/gs/enroll/62
/sjtu/gs/train
/sjtu/gs/degree
/sjtu/gs/exchange/67
/sjtu/gs/exchange/68
/sjtu/gs/exchange/69
/sjtu/gs/exchange/70
/sjtu/gs/exchange/71
/sjtu/gs/xsjy
```

## 新 RSS 路由检查表 / New RSS Route Checklist
  
- [ ] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [x] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [x] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note

dzx-dzx committed:
fix(route/sjtu): gs

dzx-dzx committed:
Correct "Incomplete URL substring sanitization"

TonyRL commented on lib/v2/sjtu/gs.js:
```diff
@@ -0,0 +1,55 @@
+                });
+                const content = cheerio.load(detailResponse.data);
+
+                item.description = (new URL(item.link).hostname === 'mp.weixin.qq.com' ? content('#js_content') : content('.page-content')).html();
```
There's a special utility to help you parse articles from `mp.weixin.qq.com` [lib/utils/wechat-mp.js](https://github.com/DIYgod/RSSHub/blob/master/lib/utils/wechat-mp.js)

TonyRL replied:
There's a special utility to help you parse articles from `mp.weixin.qq.com` [lib/utils/wechat-mp.js](https://github.com/DIYgod/RSSHub/blob/master/lib/utils/wechat-mp.js)

TonyRL commented on lib/v2/sjtu/radar.js:
```diff
@@ -0,0 +1,13 @@
+            {
+                title: '研究生通知公告',
+                docs: 'https://docs.rsshub.app/university.html#shang-hai-jiao-tong-da-xue-yan-jiu-sheng-tong-zhi-gong-gao',
+                source: ['*'],
+                target: '/gs/:type/:num?',
```
```suggestion
                source: ['/announcement/:type'],
                target: '/gs/:type/:num?',
```

TonyRL commented on docs/university.md:
```diff
@@ -2590,11 +2590,23 @@ jsjxy.hbut.edu.cn 证书链不全，自建 RSSHub 可设置环境变量 NODE_TLS
 ### 研究生通知公告
 
-<Route author="mzr1996" example="/sjtu/gs/tzgg/pyxx" path="/sjtu/gs/tzgg/:type?" :paramsDesc="['默认列举所有通知公告']">
+<Route author="dzx-dzx" example="/sjtu/gs/enroll/59" path="/sjtu/gs/:type/:num?" :paramsDesc="['类别', '细分类别, 仅对`type`为`enroll`或`exchange`有效']">
```
Do not remove the name of previous author(s)

dzx-dzx replied:
But…Wouldn't it be awkward that for every issue created for this route, the previous author will get pinged for the code he has no knowledge of?

TonyRL replied:
Pinging depends on the entries in `maintainer.js`.

dzx-dzx replied:
Oh OK. Thanks! It's now fixed in https://github.com/DIYgod/RSSHub/pull/12758/commits/b236b25f1f3b05e7ef3528c35a7612dc350590ec .

dzx-dzx committed:
Use built-in etchArticle for Wechat article.

dzx-dzx committed:
Update lib/v2/sjtu/radar.js

Co-authored-by: Tony <TonyRL@users.noreply.github.com>

dzx-dzx committed:
Fix missing semicolon.

dzx-dzx committed:
Update university.md

TonyRL committed:
refactor: migrate to v2

rename maintainer GitHub ID from `richardchien` to `stdrc`
https://github.com/DIYgod/RSSHub/pull/4064

TonyRL committed:
fix(radar): add default case
